### PR TITLE
Extract boost smt and remove unused tests

### PR DESCRIPTION
**Pull request by leonardoalt**
The SMTCheckerJSON tests should be done on the `solc-js` tests side, and the only 2 left SMTChecker Boost tests were extracted.

### CODE REVIEW

**Comment by axic** [comment]
Don't we even have a smoketest for the `auxiliaryInput`?

**Comment by leonardoalt** [comment]
We used to

**Comment by axic** [comment]
Where is it now?

**Comment by leonardoalt** [comment]
In the solc-js tests.

**Comment by axic** [comment]
I think ideally there would be a smoke tests here, and leaving the rest to solc-js. Btw, why was it moved there?

**Comment by leonardoalt** [comment]
Because it's annoying to do it here. We removed at some point because z3's nondet was affecting it and we couldn't solve it. It wasn't just smoke, there was a simple assertion in the test.

File: test/libsolidity/smtCheckerTests/imports/import_base.sol [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+pragma experimental SMTChecker;
+import "base";
+contract Der is Base {
+	function g(uint y) public {
+		x += f();
+		assert(y > x);
+	}
+}
+// ----
+// Warning 1218: (der:101-109): Error trying to invoke SMT solver.
```
Comment (chriseth) [inline test/libsolidity/smtCheckerTests/imports/import_base.sol:21]: Is this expected?

Comment (leonardoalt) [reply test/libsolidity/smtCheckerTests/imports/import_base.sol:21]: Can happen when CHC checks for underflow/overflow here. I still really dislike this message, but z3 throws when `rlimit` is over...

**Review by chriseth — APPROVED** [review]

**Pull request merged**